ls1intum/Artemis
# `Plagiarism checks`: Improve usability of similarity distribution

### Checklist
#### General


#### Client
### Motivation and Context


The idea of this PR is to increase the usability of plagiarism detection in some details. It is based on the feedback by @dveljacic and @ole-ve for #4669.
### Description

The changes of this PR are rather minor. 
First, it changes the workflow of the plagiarism detection a bit. After running the plagiarism detection, not the first comparison case is shown but the similarity distribution. This allows the instructor to get a first overview about the whole situation. The chart also allows filtering for a specific similarity interval and provides information about how many cases are already processed for one interval.
In addition to that, we disable the confirm/deny button now if the case already has been confirmed/denied. This helps the instructor to keep an overview about already processed cases. 
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 {Modeling, Text, Programming} Exercise with similar submissions from different users

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the exercise detail page of the prepared exercise
4. Click 'Check plagiarism'
5. Make sure that the chart is shown after the plagiarism check is finished
6. Select a comparison case and confirm it. Make sure that the confirm button is now disabled
7. Select another comparison case and deny it. Make sure that the deny button is now disabled
8. Return to the chart by clicking "Run details"
9. Hover over the bars of the intervals corresponding to the cases you confirmed and denied. Make sure that the distribution of confirmed/denied/open cases is updated appropriately.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


### Example workflow:

https://user-images.githubusercontent.com/80622272/157880631-769d87dc-7674-49eb-b116-07f69488c777.mp4
